### PR TITLE
feat(sage-monorepo): update PR template (SMR-754)

### PR DESCRIPTION
**Pull request by sagely1**
## Description
Updates the repository's default pull request template to add a **Testing** section where authors can suggest concrete steps and commands reviewers can run to verify the change.

## Related Issue

[SMR-754](https://sagebionetworks.jira.com/browse/SMR-754)

## Changelog

- Add `## Testing` section to `.github/pull_request_template.md` with guidance comments (manual steps, automated test commands, edge cases)
- Updates the `#(issue)` placeholder from the Related Issue section

[SMR-754]: https://sagebionetworks.jira.com/browse/SMR-754?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -10,8 +10,6 @@ the changes in the section Changelog (see below). -->
 <!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
 <!--- Please link to the issue here: -->
 
-Fixes #(issue)
-
```
Comment (hallieswan) [inline .github/pull_request_template.md:14]: Should we still include the bullet points here to make it clear that multiple issues can be listed?

**Review by hallieswan — CHANGES_REQUESTED** [review]
Looks good, can we also update the pr-description skill to align with these changes?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -24,3 +22,10 @@ Fixes #(issue)
 <!-- Demonstrate the feature or bug fix implemented in this PR. For example, -->
 <!-- - a screencast that illustrates a new UI feature (e.g. using https://gifcap.dev) -->
 <!-- - a request example and its response from a new/updated REST API endpoint -->
+
+## Testing
+
+<!-- Suggest tests reviewers can run to verify correct behavior. For example: -->
+<!-- - Steps to manually exercise the feature or bug fix -->
+<!-- - Commands to run the relevant automated tests (e.g. `nx test <project>`) -->
```
Comment (rrchai) [inline .github/pull_request_template.md]: This may not be necessary, since the affected projects will be covered by PR CI tests.

For the “Testing” section, feel free to propose a format that works best for you.

From my perspective:

- It would be helpful to include CLI commands to start the app/server, depending on whether it’s a frontend or backend change.
- For frontend work, as you mentioned in the meeting, a comparison table with before/after screenshots would be a great addition to the template.
- For backend changes, I’m not sure there’s a better approach than including example request URLs along with JSON responses, but open to other ideas if you have them.

Some of them might be overlapped with Preview section we already have.

**Review by rrchai — CHANGES_REQUESTED** [review]
LGTM overall! Thanks  - I have one suggestion.

File: .github/pull_request_template.md (hunk 2 above)
Comment (sagely1) [reply .github/pull_request_template.md]: > For frontend work, as you mentioned in the meeting, a comparison table with before/after screenshots would be a great addition to the template.

Perhaps it makes more sense to remove Preview since it might not be obvious or there may not be a visual section?  I will remove the Preview section and put some boilerplate in the Testing section.  Since we've been using a Before / After markdown table frequently, I'll include it in the boilerplate.

File: .github/pull_request_template.md (hunk 1 above)
Comment (sagely1) [reply .github/pull_request_template.md:14]: I haven't been using the Fixes prefix since the title usually calls out if there is a bug fix.  I can put a list of issues.  Let me know if it is clearer this way.

File: .claude/skills/pr-description/SKILL.md [hunk 3]
```diff
@@ -105,18 +103,20 @@ The [component/service/application] now has:
...
-- Highlight key improvements, features, or fixes
-- Mention metrics if applicable (test coverage, performance)
-- Aim for 3-5 bullet points describing major accomplishments
+- ALWAYS propose concrete verification steps — never leave this section as a placeholder or comment-only block
+- Base every suggestion on the actual diff; do not emit generic boilerplate
+- Cover three angles when relevant:
+  1. **Manual steps** — how a reviewer exercises the feature or bug fix (golden path + notable edge cases)
+  2. **Automated commands** — the exact commands for affected projects, e.g. `nx test <project>`, `nx lint <project>`, `nx e2e <project>`, `./gradlew :<project>:test`, `uv run pytest`
+  3. **Regressions / adjacent behavior** — related flows worth spot-checking to catch unintended impact
+- Use flat, single-level bullets with dashes (-); no nested items, no emojis
```
Comment (hallieswan) [inline .claude/skills/pr-description/SKILL.md]: Can you provide example output from using this skill? I'm wondering whether it would be useful for the steps for each "angle" to be in its own section

Comment (sagely1) [reply .claude/skills/pr-description/SKILL.md]: > Can you provide example output from using this skill? I'm wondering whether it would be useful for the steps for each "angle" to be in its own section

I think I have some more ideas for further refining the skill - I'll create a meeting for it next week to gather feedback on what the team wants.  This will just be a small initial step to refining the skill.

The output (based on this PR) is below:

```
## Description

Replace the PR template's "Preview" section with a "Testing" section so authors are prompted to give reviewers concrete, runnable verification steps rather than a demo-only preview. The visual Before/After table is retained as an optional note for PRs with a UI aspect, so nothing is lost for visual changes. The companion `pr-description` Claude skill is updated in lockstep so auto-generated PR descriptions emit the new Testing shape.

## Related Issue

- [SMR-754](https://sagebionetworks.jira.com/browse/SMR-754)

## Changelog

- Replace the "Preview" section in the PR template with a "Testing" section that prompts for manual steps, automated test commands, and regressions to watch
- Keep a commented-out Before/After table under Testing as an optional aid for visual changes
- Switch the Related Issue block from a single `Fixes #(issue)` line to a bulleted list so multiple issues can be linked
- Rework the `pr-description` skill so generated descriptions emit a Testing section grounded in the actual diff, never a placeholder
- Tighten the skill's quality checklist to forbid emojis, nested bullets, and boilerplate Testing content

## Testing

- Open a draft PR on any branch and confirm the GitHub "write" tab renders the new template with a **Testing** heading (and no **Preview** heading)
- Confirm the Related Issue section now shows a bulleted list placeholder instead of `Fixes #(issue)`, and that linking two issues renders both as clickable references
- Verify the commented Before/After table under Testing stays hidden in the rendered preview until uncommented
- Run the `pr-description` skill against a small sample diff in this repo and confirm the output has a Testing section with concrete steps referencing real projects from the diff (no placeholders, no emojis, no nested bullets)
- Spot-check that `.github/workflows/lint-pr.yml` still passes for a conventional title using one of the allowed types/scopes (`docs(sage-monorepo): ...`) — no changes expected here, but worth confirming since the template is adjacent
```

File: .claude/skills/pr-description/SKILL.md [hunk 4]
```diff
@@ -105,18 +103,20 @@ The [component/service/application] now has:
...
-- Use simple bullet points with dashes (-)
-- Focus on end results and benefits from user/system perspective
-- Highlight key improvements, features, or fixes
-- Mention metrics if applicable (test coverage, performance)
-- Aim for 3-5 bullet points describing major accomplishments
+- ALWAYS propose concrete verification steps — never leave this section as a placeholder or comment-only block
+- Base every suggestion on the actual diff; do not emit generic boilerplate
+- Cover three angles when relevant:
+  1. **Manual steps** — how a reviewer exercises the feature or bug fix (golden path + notable edge cases)
+  2. **Automated commands** — the exact commands for affected projects, e.g. `nx test <project>`, `nx lint <project>`, `nx e2e <project>`, `./gradlew :<project>:test`, `uv run pytest`
```
Comment (rrchai) [inline .claude/skills/pr-description/SKILL.md]: I recommend removing automated testing from this skill, as it is intended only for generating PR descriptions. Most of the relevant automatic checks are already covered by CI pipelines or hooks, ~~and running them here could unnecessarily slow down PR description creation if AI would start running them first, particularly when root-level files are modified~~. Or clarify the examples of commands.

Including manual steps, on the other hand, would be a useful addition.

**Review by hallieswan — APPROVED** [review]

**Review by rrchai — CHANGES_REQUESTED** [review]
Just one suggestion to clarify the scope to focus on adding manual steps for testing during the PR description creation. Otherwise look good.

File: .claude/skills/pr-description/SKILL.md (hunk 4 above)
Comment (rrchai) [reply .claude/skills/pr-description/SKILL.md]: I just tested this skill with a large PR (#4010), and it seems that those automated commands are not being run under the hood, as the PR description was generated fairly quickly.

Screenshot 2026-04-23 at 3 32 32 PM


It might be helpful to clarify that these automated CLI checks are not already handled by CI or existing hooks, so reviewers do not need to re-run them manually.

Comment (sagely1) [reply .claude/skills/pr-description/SKILL.md]: Thanks for the suggestions and testing the output.  I will back out changes to this skill as it is already out of scope.  We will refine this further in https://sagebionetworks.jira.com/browse/SMR-755

**Review by rrchai — APPROVED** [review]
